From 5508c816d32d5e7c7b10dad22502202a3f85dc84 Mon Sep 17 00:00:00 2001
From: Narendra Sankar <nsankar@broadcom.com>
Date: Thu, 16 Apr 2015 14:03:54 +0800
Subject: [PATCH 37/40] Added support for new ATMEL MCU - Variant - SAM4SD32B

Signed-off-by: Narendra Sankar <nsankar@broadcom.com>
---
 src/flash/nor/at91sam4.c | 44 ++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/src/flash/nor/at91sam4.c b/src/flash/nor/at91sam4.c
index c40d085..a8f505c 100644
--- a/src/flash/nor/at91sam4.c
+++ b/src/flash/nor/at91sam4.c
@@ -695,6 +695,50 @@ static const struct sam4_chip_details all_sam4_details[] = {
 		},
 	},
 
+	/*at91sam4sd32b*/
+	{
+		.chipid_cidr    = 0x29970ee0,
+		.name           = "at91sam4sd32b",
+		.total_flash_size     = 2048 * 1024,
+		.total_sram_size      = 160 * 1024,
+		.n_gpnvms       = 3,
+		.n_banks        = 2,
+
+/*		.bank[0] = { */
+		{
+			{
+				.probed = 0,
+				.pChip  = NULL,
+				.pBank  = NULL,
+				.bank_number = 0,
+				.base_address = FLASH_BANK0_BASE_SD,
+				.controller_address = 0x400e0a00,
+				.flash_wait_states = 6,	/* workaround silicon bug */
+				.present = 1,
+				.size_bytes =  1024 * 1024,
+				.nsectors   =  128,
+				.sector_size = 8192,
+				.page_size   = 512,
+			},
+
+/*		.bank[1] = { */
+			{
+				.probed = 0,
+				.pChip  = NULL,
+				.pBank  = NULL,
+				.bank_number = 1,
+				.base_address = FLASH_BANK1_BASE_2048K_SD,
+				.controller_address = 0x400e0c00,
+				.flash_wait_states = 6,	/* workaround silicon bug */
+				.present = 1,
+				.size_bytes =  1024 * 1024,
+				.nsectors   =  128,
+				.sector_size = 8192,
+				.page_size   = 512,
+			},
+		},
+	},
+
 	/* at91samg53n19 */
 	{
 		.chipid_cidr    = 0x247e0ae0,
-- 
1.8.3.msysgit.0

